drivers/disk/Kconfig.flash

```diff
@@ -3,6 +3,7 @@
 
 config DISK_DRIVER_FLASH
 	bool "Flash"
+	default y if DT_HAS_ZEPHYR_FLASH_DISK_ENABLED
```

Comment:
```
	default y
	depends on DT_HAS_ZEPHYR_FLASH_DISK_ENABLED
```

Reply:
The other disk drivers all use the form in the PR
https://github.com/zephyrproject-rtos/zephyr/blob/f320535f84db3b9bef8c1e975aa2fe61673c95c0/drivers/disk/Kconfig.mmc#L6
https://github.com/zephyrproject-rtos/zephyr/blob/f320535f84db3b9bef8c1e975aa2fe61673c95c0/drivers/disk/Kconfig.ram#L6
https://github.com/zephyrproject-rtos/zephyr/blob/f320535f84db3b9bef8c1e975aa2fe61673c95c0/drivers/disk/Kconfig.sdmmc#L9-L10

Reply:
This is the usual pattern, feel free to fix it in Kconfig.mmc and Kconfig.ram as well. Looks like Kconfig.sdmmc would need more cleanup.

Reply:
I have updated this, but please remember that asking for additional changes is against the reviewer expectations.
It feels bad to implement something according to the existing pattern and then be told to change the pattern.

Reply:
Oh, was it too much to ask that you make your changes properly? https://github.com/zephyrproject-rtos/zephyr/pull/75753#discussion_r1673636391